packages/core/src/config.ts

```diff
@@ -82,12 +82,12 @@ export function resolveConfig<Theme extends {} = {}>(
     .filter(Boolean)
     .reverse() as ResolvedConfig<Theme>['rulesDynamic']
 
-  const theme: Theme = clone([
-    ...sortedPresets.map(p => p.theme || {}),
-    config.theme || {},
-  ].reduce<Theme>((a, p) => mergeDeep(a, p), {} as Theme))
-
-  ;(mergePresets('extendTheme') as ThemeExtender<any>[]).forEach(extendTheme => extendTheme(theme))
+  let theme = [
+    ...sortedPresets.map(p => p.theme),
+    config.theme,
+  ].reverse().find(Boolean) as Theme
+  theme = (mergePresets('extendTheme') as ThemeExtender<Theme>[])
+    .reduce((mergedTheme, extendTheme) => mergeDeep(mergedTheme, extendTheme(mergedTheme) || {}), theme)
```

Comment:
I think the logic here is wrong, you need to consider the existence of other preset theme configurations.

And I switched to your branch and did a simple test and found that none of the results were what I wanted.[https://github.com/Dunqing/unocss/pull/1](https://github.com/Dunqing/unocss/pull/1).

And you'd better add more test case for it, thanks.

Reply:
Thank you for your review. 

In my opinion the result what i wanted. Because i think `preset` should respect this rule. most presets should not set `theme`, but rather use `extendTheme` to expand the theme.

I will add test cases later to show the intent of this change.